scripts/core/EXP-COMMAND-BUFER.rst

```diff
@@ -0,0 +1,116 @@
+<%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+.. _experimental-command-buffer:
+
+==============
+Command-Buffer
+==============
+
+.. warning::
+
+    Experimental features:
+
+    *   May be replaced, updated, or removed at any time.
+    *   Do not require maintaining API/ABI stability of their own additions
+        over time.
+    *   Do not require conformance testing of their own additions.
+
+
+A command-buffer represents a series of commands for execution on a command queue.
+Many adapters support this kind of construct either natively or through extensions,
+but they are not available to use directly. Typically their use is abstracted through
+the existing Core APIs, for example when calling ${x}EnqueueKernelLaunch the adapter
+may both append the kernel command to a command-buffer-like construct and also submit
+that command-buffer to a queue for execution. These types of structures allow for
+batching of commands to improve host launch latency, but without direct control it
+falls to the adapter implementation to implement automatic batching of commands.
+
+This experimental feature exposes command-buffers in the Unified Runtime API directly,
+allowing applications explicit control over the enqueue and execution of commands to
+batch commands as required for optimal performance.
+
+Querying Command-Buffer Support
+===============================
+
+Support for command-buffers can be queried for a given device/adapter by using the
+device info query with ${X}_DEVICE_INFO_EXTENSIONS. Adapters supporting this experimental
+feature will report ${X}_COMMAND_BUFFER_EXTENSION_STRING_EXP in the returned list of
+supported extensions.
+
+.. parsed-literal::
+
+    // Retrieve length of extension string
+    size_t returnedSize;
+    ${x}DeviceGetInfo(hDevice, ${X}_DEVICE_INFO_EXTENSIONS, 0, nullptr, &returnedSize);
+
+    // Retrieve extension string
+    std::unique_ptr<char[]> extensionString(new char[returnedSize]);
+    ${x}DeviceGetInfo(hDevice, ${X}_DEVICE_INFO_NAME, returnedSize, extensionString.get(), nullptr);
+    
+    std::string_view ExtensionsString(Result.get());
+    bool CmdBufferSupport = ExtensionsString.find(${X}_COMMAND_BUFFER_EXTENSION_STRING_EXP) != std::string::npos;
```

Comment:
Given this is documentation the goal should be clarity rather than exactly what `clang-format` would produce. If tooling does output suitable stuff then great but I mainly want to avoid having to scroll sideways in code boxes in the rendered HTML.

Reply:
Sorry perhaps I wasn't clear but the above code produces an error that prevents HTML generation altogether, for example:

```
Generating ../docs/source/core/CONTRIB.rst...
Generating ../docs/source/core/EXP-COMMAND-BUFER.rst...
/home/ben/dev/intel/unified-runtime/scripts/core/EXP-COMMAND-BUFER.rst(114) : error : $xCommandBufferAppendMemcpyUSMExp parameter count mismatch - 6 actual vs. 7 expected
line =     ${x}CommandBufferAppendMemcpyUSMExp(hCommandBuffer, pDst, pSrc, size, 0,

Failed to generate specification.
FAILED: CMakeFiles/generate-code /home/ben/dev/intel/unified-runtime/build/CMakeFiles/generate-code 
cd /home/ben/dev/intel/unified-runtime/scripts && /usr/bin/python3.8 run.py --api-json /home/ben/dev/intel/unified-runtime/build/unified_runtime.json && /usr/bin/python3.8 json2src.py --api-json /home/ben/dev/intel/unified-runtime/build/unified_runtime.json /home/ben/dev/intel/unified-runtime
ninja: build stopped: subcommand failed.
FAILED: CMakeFiles/generate /home/ben/dev/intel/unified-runtime/build/CMakeFiles/generate 
cd /home/ben/dev/intel/unified-runtime/build && /usr/local/bin/cmake --build /home/ben/dev/intel/unified-runtime/build --target generate-code && /usr/local/bin/cmake --build /home/ben/dev/intel/unified-runtime/build --target cppformat
ninja: build stopped: subcommand failed.
```

Reply:
I'll have a look into the errors though, and come back to this.

Reply:
Yeah, this will need to be fixed in the generator scripts. I've opened #608 to address that bug and will resolve this now.